### PR TITLE
Fixes #7971: Non applicable reports "Not applicable is not applicable" are useless and can be duplicates, breaking reporting

### PR DESCRIPTION
**Pull request by jooooooon**
https://www.rudder-project.org/redmine/issues/7971

### CODE REVIEW

File: tests/unit/test_ncf_rudder.py [hunk 1]
```diff
@@ -125,6 +138,8 @@ def test_rudder_reporting_with_variable_content(self):
 
     generated_result = ncf_rudder.generate_rudder_reporting(self.reporting_with_var_metadata)
 
+    print(self._unidiff_output(result, generated_result))
```
Comment (ncharles) [inline tests/unit/test_ncf_rudder.py]: there is an extra print there

**Comment by jooooooon** [comment]
Commit modified

File: tests/unit/test_ncf_rudder.py [hunk 2]
```diff
@@ -125,6 +138,11 @@ def test_rudder_reporting_with_variable_content(self):
 
     generated_result = ncf_rudder.generate_rudder_reporting(self.reporting_with_var_metadata)
 
+    diff = self._unidiff_output(result, generated_result)
```
Comment (VinceMacBuche) [inline tests/unit/test_ncf_rudder.py]: Maybe you can add a comment so we knwo why do that (failing test output are impossible to understand otherwise)

**Comment by VinceMacBuche** [comment]
Appart the comment I suggest to add on the diff, you can add ready to merge label

**Comment by jooooooon** [comment]
Commit modified

**Pull request merged**